dandischema/metadata.py

```diff
@@ -119,6 +120,8 @@ def validate(obj, schema_version=None, schema_key=None):
     schema_key: str, optional
       Name of the schema key to be used, if not specified, `schemaKey` of the
       object will be consulted
+    for_post: bool, optional
+      Checks if the metadata is suitable for posting to the api server
```

Comment:
not clear from this description "and then what?"
```suggestion
      Checks if the metadata is suitable for posting to the api server. If fails validation,
      but ValidationError only includes "field required" errors, a `ValueError` is raised with
      the list of all errors.
```

and then `Raises` below to be adjusted to describe that it might be raised also in `for_post=True` mode.